.github/workflows/tests_with_coverage.yml

```diff
@@ -7,39 +7,48 @@ on:
         required: false
 
 jobs:
+  build-matrix:
+    name: Generate Build Matrix
+    uses: ./.github/workflows/matrix-python.yml
 
   Tests:
-    name: Tests (Python-${{ matrix.python-version }}, Exasol-${{ matrix.exasol-version}})
+    name: Unit-Tests (Python-${{ matrix.python-version }}, Exasol-${{ matrix.exasol-version}})
+    needs: [ build-matrix ]
     runs-on: ubuntu-latest
     env:
       GITHUB_TOKEN: ${{ secrets.ALTERNATIVE_GITHUB_TOKEN ||  secrets.GITHUB_TOKEN }}
     strategy:
       fail-fast: false
-      matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
-        exasol-version: ["7.1.9"]
+      matrix: ${{ fromJson(needs.build-matrix.outputs.matrix) }}
 
     steps:
       - name: SCM Checkout
         uses: actions/checkout@v4
 
       - name: Setup Python & Poetry Environment
-        uses: exasol/python-toolbox/.github/actions/python-environment@0.17.0
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.20.0
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: Run Unit Tests
-        run: poetry run nox -s test:unit
+      - name: Run Tests and Collect Coverage
+        run: poetry run nox -s test:unit -- -- --coverage
 
       - name: Upload Artifacts
         uses: actions/upload-artifact@v4.4.0
         with:
-          name: coverage-python${{ matrix.python-version }}
+          name: coverage-python${{ matrix.python-version }}-fast
           path: .coverage
           include-hidden-files: true
 
+  Db-Version-Minimal-Tests:
+    uses: ./.github/workflows/test_db_versions_minimal.yml
+  Test-Docker-Starter:
+    uses: ./.github/workflows/test_docker_starter.yml
+  Test-Shell-Scripts:
+    uses: ./.github/workflows/test_shell_scripts.yml
   Slow-Tests:
     name: Slow
+    if: github.event_name != 'pull_request'
```

Comment:
This is not ideal, but `ci.yml` is also executed by a CRON schedule, and it would hang otherwise. I'am open for suggestions.

Reply:
What do you mean with hang?

Reply:
And the condition not pull request would deactivate this job in pull requests events. cron is a different event.

Reply:
What we actually want is the deactivate the approval for automated checks like cron or push to main

Reply:
I think, that is probably only possible if we split up the tests into two workflow files, one for fast and one for slow and add the manual approval in the caller and for the automated stuff we have a different caller as for the PR CI. I see iteration three coming of the plan